### PR TITLE
zec: implement zip 317

### PR DESCRIPTION
**Pull request by buck54321**
Adds support for ZIP-0317 fee calculations, though we've always done fees for Zcash wrong. Previously, fee should have been 1000 zats per tx, regardless of size. Now, fees are 5000 zats per "logical action".

Making these calculations possible required a pretty deep integration. I've tried to do so in a way that makes this easy to review. This has been roughly tested manually.

### CODE REVIEW

File: server/asset/zec/zec.go [hunk 1]
```diff
@@ -148,6 +151,39 @@ func (be *ZECBackend) Contract(coinID []byte, redeemScript []byte) (*asset.Contr
...
+	}
+
+	return fees >= zecTx.TxFeesZIP317()
+}
+
+// CalcOrderFunds is the ZIP-0317 compliant version of calc.RequiredOrderFunds.
+// Satisfies the asset.OrderEstimator interface.
+func (be *ZECBackend) CalcOrderFunds(swapVal, inputCount, inputsSize, maxSwaps uint64) uint64 {
+	return dexzec.RequiredOrderFunds(swapVal, inputCount, inputsSize, maxSwaps)
+}
```
Comment (martonp) [inline server/asset/zec/zec.go:185]:
``` golang
var _ asset.OrderEstimator = (*ZECBackend)(nil)
```

**Review by martonp — COMMENTED** [review]
It's looking good to me. Still need to test though.

File: docs/images/zip-0317.tex [hunk 2]
```diff
@@ -0,0 +1,20 @@
+\documentclass[]{article}
```
Comment (chappjc) [inline docs/images/zip-0317.tex:1]: Why don't we just link to the ZIP 317 page?  I don't think we need to check in a 120 KB png file either.

Comment (buck54321) [reply docs/images/zip-0317.tex:1]: Ooooh. I was reading at https://github.com/zcash/zips/blob/main/zip-0317.rst, where the tex doesn't render

File: client/asset/btc/btc.go [hunk 3]
```diff
@@ -4951,11 +5031,10 @@ var dummyP2PKHScript = []byte{0x76, 0xa9, 0x14, 0xe4, 0x28, 0x61, 0xa,
 
 // EstimateSendTxFee returns a tx fee estimate for sending or withdrawing the
 // provided amount using the provided feeRate.
-func (btc *intermediaryWallet) EstimateSendTxFee(address string, sendAmount, feeRate uint64, subtract bool) (fee uint64, isValidAddress bool, err error) {
+func (btc *baseWallet) EstimateSendTxFee(address string, sendAmount, feeRate uint64, subtract bool) (fee uint64, isValidAddress bool, err error) {
```
Comment (buck54321) [inline client/asset/btc/btc.go:5034]: This makes the electrum wallets `TxFeeEstimator`s. Anything wrong with that @chappjc?

Comment (chappjc) [reply client/asset/btc/btc.go:5034]: Not in principle.  With your move of `estimateSendTxFee` to be a `baseWallet` method, it looks like the existing `(*electrumWallet).listUnspent` is adequate.

**Comment by buck54321** [comment]
Replaced by #2553

**Pull request closed without merging**